crates/rattler_package_streaming/src/write.rs

```diff
@@ -137,22 +137,35 @@ pub fn write_tar_bz2_package<W: Write>(
 /// Write the contents of a list of paths to a tar zst archive
 fn write_zst_archive<W: Write>(
     writer: W,
+    tar_path: PathBuf,
     base_path: &Path,
     paths: impl Iterator<Item = PathBuf>,
     compression_level: CompressionLevel,
     timestamp: Option<&chrono::DateTime<chrono::Utc>>,
 ) -> Result<(), std::io::Error> {
-    // TODO figure out multi-threading for zstd
-    let compression_level = compression_level.to_zstd_level()?;
-    let mut archive = tar::Builder::new(zstd::Encoder::new(writer, compression_level)?);
+    // Create a temporary tar file
+    let tar_file = File::create(&tar_path)?;
+    let mut archive = tar::Builder::new(tar_file);
     archive.follow_symlinks(false);
-
     for path in paths {
         append_path_to_archive(&mut archive, base_path, &path, timestamp)?;
     }
+    archive.finish()?;
 
-    archive.into_inner()?.finish()?;
+    // Compress it as tar.zst
+    let mut tar_file = File::open(&tar_path)?;
+    let compression_level = compression_level.to_zstd_level()?;
+    let mut zst_encoder = zstd::Encoder::new(writer, compression_level)?;
+    zst_encoder.multithread(num_cpus::get() as u32)?;
```

Comment:
I think this should be configurable (default to this number, but users can choose lower number if they want).

Reply:
Added `ZSTD_NUMTHREADS` variable in 6ae2197c5c7f677b0632dd7e6379d796b3084679

The inspiration came from: https://github.com/facebook/zstd/issues/2241

I added a doc comment to the function but it is private thus not very visible. I think we can mention this somewhere, where would be a good place? I thought about putting it under a new section as "Environment Variables" in README.md where we list the other variables as well (I think there is only `CONDA_PREFIX` for now).